packages/transaction-controller/CHANGELOG.md

```diff
@@ -5,8 +5,55 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [17.0.0]
+### Added
+- **BREAKING:** Add swaps logic into `TransactionController` ([#1877](https://github.com/MetaMask/core/pull/1877))
+  - Swap transaction updates can be prevented by setting `disableSwaps` as `true`. If not set it will default to `false`.
+  - This value MUST be set to true in mobile, since mobile doesn't use any swap logic.
+  - If `disableSwaps` is `false` or not set, then the `createSwapsTransaction` callback MUST be defined.
+- Add hooks to support mmi logic ([#1787](https://github.com/MetaMask/core/pull/1787))
+  - Add the option in the construction `hooks` giving the possibility to pass callback functions additional logic to execute.
+  - Add the optional callback in hooks `getAdditionalSignArguments` returns additional arguments required to sign a transaction.
+  - Add the optional callback in hooks `beforeApproveOnInit` additional logic to execute before starting an approval flow for a transaction during initialization. Return false to skip the transaction.
+  - Add the optional callback in hooks `afterSign` additional logic to execute after signing a transaction. Return false to not change the status to signed.
+  - Add the optional callback in hooks `beforePublish` additional logic to execute before publishing a transaction. Return false to prevent the broadcast of the transaction.
```

Comment:
```suggestion
  - Add the `beforePublish` hook to execute additional logic before publishing a transaction. Return `false` to prevent the transaction being submitted.
```